inception-project/inception
# #1054 - Dealing with many results in search sidebar

**What's in the PR**
- [x] integrate paging directly in the query
- [x] implement wicket Dataprovider and Dataview for search results
- [x] adjust SearchService and Sidebar to work with the paging
- [x] Hide page navigator if there are no search results yet 

When no grouping is activated (i.e. default grouping by document) and "low level paging" Checkbox is checked, paging will be applied at "query-level" (MtasDocumentIndex). 

Otherwise, all results are fetched at once and paging is applied for the rendering. 

This is done because if we apply paging at query level and grouping together, members of a
group might me scattered over multiple pages. (Grouping by document is an exception because the query iterates over documents. So results from the same document appear in a sequence)

**How to test manually**
* Make queries and annotations in the search sidebar, see if it still works. Selections of search results should persists when switching between different pages.
* Test queries with a large number of results, see if it is reasonably fast.

**Automatic testing**

**Documentation**
